pennylane_lightning/core/src/algorithms/CMakeLists.txt

```diff
@@ -15,7 +15,7 @@ target_link_libraries(lightning_algorithms INTERFACE    lightning_compile_option
                                                         lightning_observables
                                                         )
 
-if (BUILD_TESTS)
+if (BUILD_TESTS AND NOT DEFINED PL_TENSOR)
```

Comment:
The reason is that we can't apply tests here for the `lightning.tensor` backend. There will be compile errors due to classes (Observables are not implemented yet) and methods (no `getData()` method in `lightning.tensor` backend) that are unavailable.

Reply:
So this is just temporary until everything is fully completed?

Reply:
I would say it's a temporary solution and further refactoring is required to remove it. Will discuss why in our meeting.

Reply:
Hey @mlxd , It's temporary for now. Another point forgot to mention is that adding PL_TENSOR is just to skip the C++ unit tests for the alogrithm, observable and measurement class instead of those classes themselves. It makes sense for now since some methods tests in those base classes are not implemented in the lightning.tensor backends. Also adding PL_TENSOR here is to resolve the potential issue of the previous implementation (conditionally skip compiling some base classing for lightning.tensor), which will prevent us from installing lightning.gpu and lightning.tensor simultaneously.